src/capi.rs

```diff
@@ -0,0 +1,502 @@
+//! # C API for rav1e
+//!
+//! [rav1e](https://github.com/xiph/rav1e/) is an [AV1](https://aomediacodec.github.io/av1-spec/)
+//! encoder written in [Rust](https://rust-lang.org)
+//!
+//! This is the C-compatible API
+
+use std::slice;
+use std::sync::Arc;
+
+use std::ffi::CStr;
+use std::ffi::CString;
+use std::os::raw::c_char;
+use std::os::raw::c_int;
+
+use libc::size_t;
+use libc::ptrdiff_t;
+
+use num_derive::*;
+use num_traits::cast::FromPrimitive;
+
+use crate::prelude as rav1e;
+
+/// Raw video Frame
+///
+/// It can be allocated throught rav1e_frame_new(), populated using rav1e_frame_fill_plane()
+/// and freed using rav1e_frame_unref().
+pub struct Frame(Arc<rav1e::Frame<u16>>);
+
+#[repr(C)]
+#[derive(Copy, Clone, Debug, FromPrimitive)]
+pub enum EncoderStatus {
```

Comment:
Are you happy with an enum here? AFAIK it's more "C-like" just to use ints for errors.

Reply:
It did work well enough for the last months so I wouldn't change it now :)

Reply:
I'm fine with enums. I like to be explicit, myself

Reply:
k